provider/terraform/virtual_fields.rb

```diff
@@ -23,6 +23,16 @@ class Terraform < Provider::AbstractCore
     # behaviour. They often don't map to underlying API fields (although they
     # may map to parameters), and will require custom code to be added to
     # control them.
+    #
+    # Virtual fields only support top level, boolean Terraform fields
+    #
+    # Virtual fields are similar to url_param_only fields in that they create
```

Comment:
I guess I'm not seeing the same connection between the two. They both add fields that have around the same end behaviour in terms of requests, but the distinction between API parameter vs TF-only field feels clear to me. I'd rather we improve that than talk extensively about implementation details. 

https://github.com/GoogleCloudPlatform/magic-modules/pull/3125/files#diff-b6834071272892b9e8214d96c8b074caR23 is my bad, I'd forgotten that you could use `url_param_only` for parameters not in the `self_link`/`base_url`.